.github/actions/orca/orca-python-ray-py37-spark3-action/nightly-test/action.yml

```diff
@@ -9,6 +9,24 @@ runs:
         source activate py37
         export SPARK_LOCAL_HOSTNAME=localhost
         pip install -i https://pypi.org/simple --pre --upgrade bigdl-orca-spark3
+        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} tensorflow==1.15.0
+        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} tensorflow-gan==2.0.0
+        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} tensorflow-probability==0.7.0
+        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} tensorflow-datasets==3.2.0
+        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} ray[default]==1.9.2
+        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} protobuf==3.15.3
+        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} torch==1.7.1
+        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} torchvision
+        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} torchmetrics
+        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} pyarrow==4.0.1
+        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} mxnet==1.6.0
+        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} mmcv
+        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} pandas
+        pip install -i ${GONDOLIN_PIP_MIRROR} --trusted-host ${GONDOLIN_TRUSTED_HOST} fsspec==2021.9.0
```

Comment:
pyarrow:
~~~
/opt/conda/envs/py37/lib/python3.7/site-packages/bigdl/orca/data/image/utils.py:22: in <module>
    import pyarrow as pa
E   ModuleNotFoundError: No module named 'pyarrow'
~~~
test link
https://github.com/intel-analytics/BigDL/actions/runs/3879774325/jobs/6617474614

Reply:
https://github.com/intel-analytics/BigDL/pull/7269 The issue with pyarrow has fixed. Merge with main and run again to see if pyarrow is still needed?

Reply:
still need pyarrow
~~~
Running orca data tests
>                       format(target_id, ".", name), value)
E                   py4j.protocol.Py4JJavaError: An error occurred while calling z:org.apache.spark.api.python.PythonRDD.collectAndServe.
E                   : org.apache.spark.SparkException: Job aborted due to stage failure: Task 2 in stage 77.0 failed 1 times, most recent failure: Lost task 2.0 in stage 77.0 (TID 187) (localhost executor driver): org.apache.spark.api.python.PythonException: Traceback (most recent call last):
E                     File "/opt/conda/envs/py37/lib/python3.7/site-packages/pyspark/python/lib/pyspark.zip/pyspark/worker.py", line 604, in main
E                       process()
E                     File "/opt/conda/envs/py37/lib/python3.7/site-packages/pyspark/python/lib/pyspark.zip/pyspark/worker.py", line 596, in process
E                       serializer.dump_stream(out_iter, outfile)
E                     File "/opt/conda/envs/py37/lib/python3.7/site-packages/pyspark/python/lib/pyspark.zip/pyspark/serializers.py", line 259, in dump_stream
E                       vs = list(itertools.islice(iterator, batch))
E                     File "/opt/conda/envs/py37/lib/python3.7/site-packages/bigdl/orca/data/utils.py", line 490, in farrow
E                       import pyarrow as pa
E                   ModuleNotFoundError: No module named 'pyarrow'
E                   
~~~
test link
https://github.com/intel-analytics/BigDL/actions/runs/3928126014/jobs/6715418478

Reply:
will look into it

Reply:
@sgwhat Could you take a look at this?
Seems `File "/opt/conda/envs/py37/lib/python3.7/site-packages/bigdl/orca/data/utils.py", line 490, in farrow` if pyarrow fails, it will fall back to without arrow optimization? Why the test will directly throw an error?

Reply:
yes these should fall back to `to_pandas_without_arrow` directly, that's wired, still look into it.

Reply:
seems call `spark_df_to_pd_rdd` is fine, but `spark_df_to_pd_xshards` and other functions which call `_to_rdd` will meet this error.

Reply:
When calling `self._get_class_name()`, most cases require installing `pyarrow`.

Reply:
Seems for shards of pandas dataframe, we use pyarrow to infer the schema. For inputs like SparkDataFrame, pyarrow won't be needed.
We can add pyarrow in the environment at this moment. And afterwards construct a job for the tests that totally don't need pyarrow to avoid the strict dependency.